frontend/src/controllers/UserKit.js

```diff
@@ -0,0 +1,41 @@
+export default function loadUserKit(appId, initFn, signInFn) {
+    const widgetJsUrl = "https://widget.userkit.io/widget.js";
+
+    let scriptEls = document.getElementsByTagName("script");
+    for (const el of scriptEls) {
+        if (el.src == widgetJsUrl) {
+            // widget.js already loaded, execute init callback
+            if (typeof initFn === 'function') {
+                initFn(window.UserKit, window.UserKitWidget);
+            }
+            // if already signed in, execute sign in callback
+            if (typeof signInFn === 'function' && window.UserKit.isLoggedIn()) {
+                signInFn(window.userKit, window.userKitWidget);
+            }
+            return;
+        }
+    }
+
+    // If callback is provided, attach a listener for 'UserKitInit'
+    if (typeof initFn === 'function') {
+        document.addEventListener("UserKitInit", () => {
+            initFn(window.UserKit, window.UserKitWidget);
+        });
+    }
+
+    // Attach listener for 'UserKitSignIn' event
+    document.addEventListener("UserKitSignIn", () => {
```

Comment:
Why do we check that `initFn` is a callback, but we don't check `signInFn` in the same way?